pages/search.js

```diff
@@ -157,7 +157,7 @@ export default function Search({
     }, 500);
 
     return () => clearTimeout(timer);
-  }, [inputValue]);
+  }, [inputValue, randUsers, BASE_URL, keyword, router]);
```

Comment:
```suggestion
  }, [inputValue, randUsers, BASE_URL, keyword]);
```
Adding `router` (useRouter) into the dependency causes the page to rerender indefinitely.

An alternative is to use `Router` (imported from "next/router") and replace `router` with `Router`. A change in line #81 is required:
```
const { username, keyword, userSearchParam } = typeof window !== 'undefined' && Router.query;
```

Reply:
Ckelwin. Thanks for your comment, I think the alternative that you suggested is probably the better choice.